server/sonar-process-monitor/src/main/java/org/sonar/process/monitor/Monitor.java

```diff
@@ -364,11 +381,14 @@ private RestartorThread() {
     public void run() {
       stopProcesses();
       try {
-        startProcesses();
+        startProcesses(Monitor.this::loadJavaCommands);
       } catch (InterruptedException e) {
         // Startup was interrupted. Processes are being stopped asynchronously.
         // Restoring the interruption state.
         Thread.currentThread().interrupt();
+      } catch (Throwable t) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1181)